packages/nextjs/src/config/loaders/ast.ts

```diff
@@ -320,3 +328,169 @@ export function removeComments(ast: AST): void {
   const nodesWithComments = ast.find(Node).filter(nodePath => !!nodePath.node.comments);
   nodesWithComments.forEach(nodePath => (nodePath.node.comments = null));
 }
+
+/**
+ * Determines from a given AST of a file, whether the file has a default export or not.
+ */
+export function hasDefaultExport(ast: AST): boolean {
+  const hasDefaultDeclaration = ast.find(ExportDefaultDeclaration).size() > 0;
+  if (hasDefaultDeclaration) {
+    return true;
+  }
+
+  const hasDefaultSpecifier = ast.find(ExportDefaultSpecifier).size() > 0;
+  if (hasDefaultSpecifier) {
+    return true;
+  }
+
+  const hasNamedDefaultExport = ast.find(ExportSpecifier).some(specifier => specifier.node.exported.name === 'default');
+  if (hasNamedDefaultExport) {
+    return true;
+  }
+
+  return false;
+}
+
+/**
+ * Extracts all identifiers from a "RestElement" within a named export declaration statement ("export const val = ...").
+ * "RestElements" are things like "...value" inside a destructuring assignment.
+ *
+ * Example:
+ * "export const { ...restElement1, bar: { ...restElement2 } } = { foo: 1, bar: { baz: 2 } }" --> ["restElement1", "restElement2"]
```

Comment:
A few questions here:

- Don't rest properties have to come last?
- Why would one gather everything in an object into a rest property? Wouldn't that just be the object itself?
- If the `RestElement` is truly just the `...others` bit, then this isn't a valid input, is it?

Reply:
1. Yup, good catch. I just barfed random examples here without using much brain power.
2. I don't know why anyone would do that either. The thing that matters here is that you *can* do that and we need to handle that case. I just put the example here to make people coming across this code section aware of such things so it puts them into the right mindset of what we need to look out for.
3. Correct, I updated the example to be a little bit more clear in what this function does

--> 145ed92912d472c33bcf3ec7df8aa502327c41c3